cpp/src/cugraph.cu

```diff
@@ -113,6 +115,105 @@ gdf_error gdf_adj_list::get_source_indices (gdf_column *src_indices) {
   return GDF_SUCCESS;
 }
 
+gdf_error gdf_renumber_vertices(const gdf_column *src, const gdf_column *dst,
+				gdf_column **src_renumbered, gdf_column **dst_renumbered,
+				gdf_column **numbering_map) {
+
+  GDF_REQUIRE( src->size == dst->size, GDF_COLUMN_SIZE_MISMATCH );
+  GDF_REQUIRE( src->dtype == dst->dtype, GDF_UNSUPPORTED_DTYPE );
+  GDF_REQUIRE( ((src->dtype == GDF_INT32) || (src->dtype == GDF_INT64)), GDF_UNSUPPORTED_DTYPE );
+  GDF_REQUIRE( src->size > 0, GDF_DATASET_EMPTY ); 
+
+  *src_renumbered = new gdf_column;
+  *dst_renumbered = new gdf_column;
+  *numbering_map = new gdf_column;
```

Comment:
So do we really need these memory allocations here? If we ignore the fact that this function is called by cython and also ignore that we are returning error code (as I know we will switch to exception based error handling), I think safer implementation is to do

```cpp
gdf_column src_renumbered;
gdf_column dst_renumbered;
gdf_column numbering_map;
...
return std::make_tuple(src_renumbered, dst_renumbered, numbering_map);
```

Now, we have our limitation and also this function needs to maintain consistency with other functions in this file. Then, I think it's better to do in cython

```
cdef gdf_column src_renumbered
cdef gdf_column dst_renumbered;
cdef gdf_column numbering_map;
...
err = gdf_renumber_vertices(..., &src_renumbered, &dst_renumbered, &numbering_map)
```

and modify this function's signature to take gdf_column* (instead of gdf_column**) for src_renumbered, dst_renumbered, and numbering_map. Then, we don't need to worry about explicitly freeing these memory allocations (and also avoid heap allocation overhead). I will comment in the cython file again, but the cython caller did not indeed free these allocations; and this leads to memory leak.

Reply:
OK.  I think I started this work before some of the cleanup you did for memory leaks.  Probably copied this from somewhere else.  My next push should address this.